dotnet/src/Functions/Functions.OpenAPI/OpenAI/KernelOpenAIPluginExtensions.cs

```diff
@@ -214,7 +214,7 @@ public static class KernelOpenAIPluginExtensions
             var callback = executionParameters.AuthCallback;
             ((OpenApiFunctionExecutionParameters)executionParameters).AuthCallback = async (request) =>
             {
-                await callback(request, pluginName, openAIAuthConfig).ConfigureAwait(false);
+                await callback(request, pluginName, openAIAuthConfig, cancellationToken).ConfigureAwait(false);
```

Comment:
The cancellation token passed to the CreateAsync is used for importing plugins and will not be valid during the plugins' execution time. To obtain the relevant cancellation token for plugin execution, it should be passed from the runner to the auth callback, which should be extended as follows: '...AuthCallback = async (request, **ct**) =>'

Reply:
good catch, fixed